### ISSUE TITLE
[Unresolved Non SteamVR-OSVR Issue] Steamvr Compositor Issue/s

### ISSUE THREAD

**Issue by Balderick** [OSVR/SteamVR-OSVR#75]
# N.B.  If You Have Been Directed Here Through OSVR Support Please read the comments from most recent backwards to try and track issues in reverse chronological order to best understand my most recent status regarding configuration being used etc.

CURRENTLY

## No Headset Detected (108)
Main current issue seems to be usb device (ir camera) firmware issue with usb 3.0 female ports.  steamvr does not detect device.  running latest current steam client, steamvr, osvr control, osvr runtime 1204 and latest osvr drivers.  Latest drivers firmware and everything else which meeans nvidia 368.81 drivers. Regardless of which osvr_server.config.json or sample file is used all that is shown in osvr server output is what is listed at https://github.com/OSVR/OSVR-Docs/issues/63

Before launching osvr server i check that all OSVR devices are showing as well and healthy in devices app, control panel device and printers as well as in device manager.

OSVR Control gives ` >#?v
Version 1.92  Mar 24 2016

Tracker:1.8.4.415 `

### SteamVR Compositor Is Not Available (400)
After receiving a HDK 1.4 kit on Thursday it took me till late on Friday to get drivers installed with no problem devices showing in device manager.  It took another twelve hours + of applying workarounds, reinstalling drivers and searching the interweb to work out i had to try rolling back from 1197 to 1194 core and steamvr-osvr drivers to improve chances of a steamvr title to launch.  My HDK display appeared to die on Saturday evening and spent most of Sunday to work out my hdk was stuck in direct mode and not dead.
The furthest in set up process i have been is launching steamvr and being asked to switch to direct mode but steamvr compositor can not start when steamvr restarts in direct mode.   Using 1197 runtime and steamvr-osvr drivers i managed to go through room setup before being asked to switch to direct mode.   

*Describe the issue you've encountered. What is the exepected and actual behavior? What steps have you tried to fix the problem?*
osvr audio device showing as problem device i.e. unusable, not started in device manager.   windows devices and devices and printers showing no ssues with displays but nvidia control panel sees hdk display as using portrait.  selecting landscape helps with steamvr issues/errors (400 and 109).  To help with steamvr setup and osvr complaining about no beacon configuration done i used https://www.youtube.com/watch?v=MObPn_U4IYg but allthough the file appeared to save osvr could never use it.   The other bugs still needing workarounds applied are described well in this video.  https://www.youtube.com/watch?v=1ZaboSUUpcI    ~~I do not expect drivers to be an issue on modern operating systems.   Killing hardware makes the drivers not fit for purpose.~~

## Steps to reproduce the problem

*How can the problem be reproduced?*
Plug in and power up HDK
Launch OSVR client
Start Steam
Start SteamVR
Switch to direct mode using steamvr ui
Can not get passed "compositor is not running (400)" dialogue.  Clicking the "launch compositor" button does nothing.  If i start steamvr with hdk in extended mode after deleting steamvr.vrsettings the compositor (white meshed environment) is visible in hdk display.    The compositor never loads after switching to direct mode.

The enable/disable direct mode switches that come with osvr runtime work as expected the majority of the time.  Roughly 3 out of 10 times get unexpected behaviour ranging from appearing to do nothing and nothing.  i.e. gets stuck in direct mode after steamvr switched from extended to direct mode.
Getting osvr audio device not be shown as trouble device in device manager is an issue in its own right but steamvr not starting compositor is what is stopping any app to launch as expected in vr mode imo.

## System configuration

*Please specify any relevant system configuration information.*

  * **Graphics card(s):** Main display: 4K TV 3840x2160   Intel 530 igpu
                                       HDK Display: 1920x1080 Asus Matrix Nvidia GTX 980TI discrete gpu
  * **Operating system:** windows 10 x64
  * **Version of SteamVR**:  Current stable
  * **Version of SteamVR-OSVR:** 1194 and 1197 tested.   Both failed to allow launching of any steamvr title.

After finding 1.2.6 and 1.2.7 drivers installed at same time i suspect what caused this to happen is same thing that has caused hdk display to die.   I double checked my downloads folder and am 100 per cent positive i did not install 1.2.6 drivers.  The only currently available drivers from https://github.com/OSVR/OSVR-HDK-Windows-Drivers/releases are 1.2.7

OSVR audio device has issues with driver management.  It is a problem device.  I need to disconnect usb from i/o on pc and use device manger to show hidden devices.  Only after deleting all disconnected devices and drivers allows the reinstall of driver for osvr audio device with no issue or problems.

## Attached files

*Please attach the following files:*

  * *The `osvr_server` configuration file.* `{
    "description": "This configuration supports video (so-called 'positional') and IMU fusion tracking, in addition to orientation-only tracking, with the OSVR HDK. It is configured for RenderManager applications in non-direct extended-display landscape mode on HDK 1.3 optics.",
    "display": "displays/OSVR_HDK_1_3.json",
    "renderManagerConfig": "sample-configs/renderManager.extended.landscape.json",
    "drivers": [{
        "plugin": "com_osvr_VideoBasedHMDTracker",
        "driver": "VideoBasedHMDTracker",
        "params": {
            "showDebug": false,
            "includeRearPanel": true,
            "headCircumference": 55.75,
            "calibrationFile": "videotrackerCombinedCalibrationFile13.json"
        }
    }, {
        "plugin": "org_osvr_filter_videoimufusion",
        "driver": "VideoIMUFusion",
        "params": {
            "name": "HeadFusion",
            "input": {
                "imu": "/com_osvr_Multiserver/OSVRHackerDevKitPrediction0/semantic/hmd",
                "faceplate": "/com_osvr_VideoBasedHMDTracker/TrackedCamera0_0/semantic/hmd/front"
            },
            "eyeHeight": 0,
            "cameraIsForward": true
        }
    }],
    "aliases": {
        "/headSpace": {
            "translate": [0.0, 0.0, 0.04141],
            "child": "/org_osvr_filter_videoimufusion/HeadFusion/semantic/fused"
        },
        "/me/head": "/headSpace"
    }
}
`
  * *The `vr*.txt` files from the Steam `logs` directory.*
which ones?.  there are a lot.
[vrclient_BrookhavenExperiment.txt](https://github.com/OSVR/SteamVR-OSVR/files/345043/vrclient_BrookhavenExperiment.txt)
[vrclient_DCS.txt](https://github.com/OSVR/SteamVR-OSVR/files/345042/vrclient_DCS.txt)
[vrclient_hellovr_sdl.txt](https://github.com/OSVR/SteamVR-OSVR/files/345047/vrclient_hellovr_sdl.txt)
[vrclient_hl2.txt](https://github.com/OSVR/SteamVR-OSVR/files/345045/vrclient_hl2.txt)
[vrclient_pCARS64.txt](https://github.com/OSVR/SteamVR-OSVR/files/345046/vrclient_pCARS64.txt)
[vrclient_ShowdownVRDemo.txt](https://github.com/OSVR/SteamVR-OSVR/files/345044/vrclient_ShowdownVRDemo.txt)
[vrclient_Steam.txt](https://github.com/OSVR/SteamVR-OSVR/files/345048/vrclient_Steam.txt)
[vrclient_steamvr_desktop_game_theater.txt](https://github.com/OSVR/SteamVR-OSVR/files/345049/vrclient_steamvr_desktop_game_theater.txt)
[vrclient_steamvr_room_setup.txt](https://github.com/OSVR/SteamVR-OSVR/files/345050/vrclient_steamvr_room_setup.txt)
[vrclient_vrcompositor.previous.txt](https://github.com/OSVR/SteamVR-OSVR/files/345052/vrclient_vrcompositor.previous.txt)
[vrclient_vrcompositor.txt](https://github.com/OSVR/SteamVR-OSVR/files/345053/vrclient_vrcompositor.txt)
[vrclient_vrdashboard.txt](https://github.com/OSVR/SteamVR-OSVR/files/345051/vrclient_vrdashboard.txt)
[vrclient_vrmonitor.previous.txt](https://github.com/OSVR/SteamVR-OSVR/files/345054/vrclient_vrmonitor.previous.txt)
[vrclient_vrmonitor.txt](https://github.com/OSVR/SteamVR-OSVR/files/345055/vrclient_vrmonitor.txt)
[vrcompositor.previous.txt](https://github.com/OSVR/SteamVR-OSVR/files/345059/vrcompositor.previous.txt)
[vrcompositor.txt](https://github.com/OSVR/SteamVR-OSVR/files/345056/vrcompositor.txt)
[vrdashboard.txt](https://github.com/OSVR/SteamVR-OSVR/files/345057/vrdashboard.txt)
[vrmonitor.previous.txt](https://github.com/OSVR/SteamVR-OSVR/files/345058/vrmonitor.previous.txt)
[vrmonitor.txt](https://github.com/OSVR/SteamVR-OSVR/files/345060/vrmonitor.txt)
[vrserver.previous.txt](https://github.com/OSVR/SteamVR-OSVR/files/345061/vrserver.previous.txt)
[vrserver.txt](https://github.com/OSVR/SteamVR-OSVR/files/345062/vrserver.txt)

## Notes
N.B. Two HDK's and 100 + hours down and still not one steamvr app launched successfuly.   :-( 

Now that i have faceplate removed am currently working my way through https://www.youtube.com/watch?v=GLgxJxbsalM to try and get going again 

Why isn't there a screwdriver for removing faceplate included with hdk?  I used a 00 phillips screwdriver with care and still rounded off a screwhead probably due to the screws being overtightened on assembly imho.

HDK was stuck in direct mode but still can not get passed "compositor not available (400)".

**Comment by Balderick** [comment]
> HDK
What is the HDK?
The HDK or Hacker Development Kit is a virtual reality HMD designed to bring the benefits of the OSVR ecosystem to fans of VR out of the box. Not only does it allow you to experience VR, its support for the OSVR ecosystem means access to the growing range of VR controllers and technologies that support the ecosystem.

There are two HDKs currently available:
- HDK 1 – Designed to provide all the essential technology you need to experience VR.
- HDK 2 – Designed to deliver mind bending visual experiences with more advanced technology.
Why is it called a Hacker Development Kit? Is it for developers or consumers?
The HDK is for both.
The HDK was designed to embody the vision of the software platform – to allow for unrestricted innovation for developers and to deliver innovation to consumers.
It was created to address the needs of the emerging VR industry that requires flexibility and technology that is open to evolution. This to help power innovation and allow owners of the hardware to access the latest technologies as soon as they are supported by the ecosystem.
While this is something we will continue if a consumer focused iteration is released, the Hacker Development Kit exists to be that perfect fit between industry innovation and the evolution of a consumer’s needs and true to the OSVR software – it is open source.

Why have i had two hdk's and never experienced vr?

**Comment by godbyk** [comment]
Do you have the HDK plugged into the Intel card or the nVidia card? It may be that SteamVR has trouble with direct mode on dual-card machines. Do things work as expected in extended mode?

**Comment by russell-taylor** [comment]
The latest nVidia drivers prevent DirectMode from operating on dual-GPU machines with non-HDCP-compliant displays.  You should be able to run in extended mode, where the laptop is on the Windows display.  You can put the display into that state using the DisableDirectMode program that should ship with the SDK.

**Comment by Balderick** [comment]
> Please specify any relevant system configuration information.

Graphics card(s): Main display: 4K TV 3840x2160 Intel 530 igpu HDK Display: 1920x1080 Asus Matrix Nvidia GTX 980TI discrete gpu

The hdk needs to be plugged into discrete card as the igpu (onboard) is not supported (intel 530) by vr   The two gpus are being used as there is only one hdmi port on nvidia discrete card.

**Comment by Balderick** [comment]
Can now confirm more unexpected behaviour.   Since receiving hdk 1.4 have very rarely put the hmd display on head due to nothing launching as expected.   When trying OSVR content from http://www.osvr.org/featured.html#page-1 not one launches as expected.

DreadEye VR demo, Showdown (direct mode), Cityscape Repairman, SightLine The Chair and The Palace Demo mentioned in documentation is not even listed as official osvr content does not laumnch as expected.
Even extended mode is not working as expected as after viewing the extended desktop in hdk display it is upside down.    
After setting landscape in nvidia control panel settings the extended desktop in hdk is upside down.  If i try to launch anything (steamvr or osvr content) in hdk and i go back to nvidia control panel the orientation has changed to portrait.    
When any osvr content is launched a stereoscopic window opens on main display.  The hdk display just stays blank due to being in direct mode.

So i got drivers that windows hates and are actualy causing wdf_violation bluescreen of death,
no VR content showing in hdk display other than a borked unusable extended desktop.

Where do i start debugging this shit?

**Comment by Balderick** [comment]
@godbyk  With refernce too https://www.reddit.com/r/OSVR/comments/4qrin6/news_osvr_runtime_more_update/d4vy4x3  please put the orientation issue higher on your priority list.

**Comment by Balderick** [comment]
How does latest nvidia drivers break osvr direct mode?

I just updated to 368.69 purely because of https://youtu.be/6jJr-DMLka8 and still hope to use hdk to enjoy     such content.

**Comment by Balderick** [comment]
Everything works as expected with 209 build steamvr-osvr drivers and current stable steamvr with 368.69 drivers up untill trying to launch an app.  The orientation issue and direct/extended mode switching isues are no longer reproducible for me.  Still can not get any steamvr app to show in hmd as expected though.
Getting wdf_violation bsod when powering down or restarting pc with hdk connected and powered up is still reproducable.


Rolling back to 365.19 drivers with current steamvr stable is a no go.

**Comment by rpavlik** [comment]
Should not need a beta of steamvr. If the hdk was "killed" that's a known hardware issue, contact support for RMA.

You might consider unplugging the USB when booting for now.

**Comment by rpavlik** [comment]
If you still have GPUs from two vendors, then rolling back to a pre 367 driver is the only solution that will help getting direct mode apps to work: you can even see the NVIDIA driver release notes at least for the first 368 release driver, they mention intentionally breaking HDK support on multi GPU systems.

**Comment by godbyk** [comment]
From the [release notes](http://us.download.nvidia.com/Windows/368.69/368.69-win10-win8-win7-desktop-release-notes.pdf) for nVidia driver version 368.69:

> With R367 and future NVIDIA drivers, Crescent Bay and Open Source Virtual Reality development kit headsets will not work with VRWorks Direct Mode in systems that contain GPUs from multiple vendors (e.g. NVIDIA and Intel). For such systems, please disable the Intel integrated graphics (from the system BIOS) in order to use Direct Mode.

**Comment by Balderick** [comment]
When i started this thread i did believe my hdk was dead.  After removing faceplate to access reset switches i noticed the tracking leds were not flashing but were on.  I then realised i should try disconnecting hdk rebooting system and reconnecting hdk.  The switches that install with runtime for enabling and disabling direct mode were functional once again.
I have since had the hdk get stuck in direct mode but simply disconnected hdk and used device manager to show hidden devices to uninstall the display adapter and then reboot and try reconnecting hdk again.  Seems to be a reliable reproducable workaround for getting unstuck from direct mode.

And again tio clarify latest gpu drivers (368.69) with 1194 and 1197 osvr runtime and steamvr-osvr drivers have no issue of switching to and from direct mode (most of the time).   It is getting apps to launch and show up in hmd that is the issue here.

**Comment by Balderick** [comment]
So this is how i read:
> With R367 and future NVIDIA drivers, Crescent Bay and Open Source Virtual Reality development kit headsets will not work with VRWorks Direct Mode in systems that contain GPUs from multiple vendors (e.g. NVIDIA and Intel). For such systems, please disable the Intel integrated graphics (from the system BIOS) in order to use Direct Mode.
 
what i see is > Direct Mode in systems that contain GPUs from multiple vendors
as meaning a gtx 950 with one hdmi port could be used to run primary display ( 4K TV in my case) paired with a gtx 980 ti for running hmd.

If my tv and hmd are hdmi only choice of connection to pc that means there has to be two available hdmi ports on pc.  

Using nvidia gtx 950 for primary display along with nvidia gtx 980 ti appears to be fine in device manager.  In nvidia control panel all the display layout, configuration and settings appear to be configured correctly.    There is no touble swtching from and to direct mode using 1197 runtime and steamvr osvr drivers with nvidia 368.69 drivers.  I now no longer even get the 400 and 109 errors as s steamvr simply does not detect hmd giving error 108.   why?   because steamvr is looking for primary and hmd connected to same gpu.   ref:  Error 208 section of https://support.steampowered.com/kb_article.php?ref=8566-SDZC-9326#headset reads: > Make sure that your Vive’s link box and the monitor or TV you use as a primary display are plugged into the same graphics card. If you are connected to your onboard display adapter, you will see an Intel HD Graphics device listed in your SteamVR system report. Onboard cards, such as those connected to a PC’s motherboard, will not be powerful enough to use the Vive. Reconnect your Vive to your PC’s dedicated graphics card.

With a dual nvidia set up with gtx 950 as primary display and gtx 980 ti as hmd due to both having only one hdmi port each does not even get close to what nvidia describes what should happen with a dual nvidia setup is not the behaviour i see with 368.69 drivers; nor with a nvidia and integrated intel gpu combination..  Direct mode switching seems to be more reliable with the intel/nvidia combo.

Windows is having issues with tracking camera now too;  which is definitely an osvr software/driver issue.  I have no usb 2.0 ports which is what windows is saying i need to try and do for tracking camera to work as expected.  


So  i opened up geforce experience and discover it is saying my pc is now not VR capable.   Why?  because my primary display and hmd are not connected to same nvidia gpu.

**Comment by Balderick** [comment]
So still trying to get hdk to behave as expected and/or as described in official documentation.  Not even trying to get steamvr apps running but just trying to get a working osvr server config working.    After recovering hdk from being stuck in direct mode once more, making sure all my mobo device drivers are up to date, making sure there are no firmware updates for any hardware on this machine available and spending 40+ hours trying to research, troubleshoot and remedy all the issues i am having with osvr hdk and osvr software.

My main focus has shifted from trying to get by steamvr errors and is now trying to work out why osvr server has stopped giving positional tracking, tracker viewer broke and can not calibrate leds by using videotrackercalibibration utility.    https://github.com/OSVR/OSVR-Docs/issues/63

Very little of the official information in osvr docs seems to be relevant to or even refer to recent builds of osvr hdk, osvr runtime and steamvr-osvr drivers .

**Comment by Balderick** [comment]
Can anybody confirm the latest runtime (1204) and the latest steamvr-osvr drivers (209--1197) works as expected with steam vr apps?

**Comment by godbyk** [comment]
I do my testing with the latest runtime and the latest (non-beta) version of SteamVR.

**Comment by Balderick** [comment]
Have found connecting a 24" 1080p monitor to same GPU as hdk hmd has indeed allowed osvr demos/apps to load in hmd but I need to switch or move primary display for different osvr demos.   The demos are not using ir positional tracking so were not very enjoyable or impressive.   
Ir positional tracking is using saved beacon colibration from videotracking configuration utility.

Too many of my questions and issues and being left unanswered and unresolved here.    Though have proven it is not my osvr hardware that is at fault.    All the issues listed are known and documented osvr software issues.

I opened another osvr support ticket so will stop asking questions and trying to communicate what issues I am having and what I am trying to do to get by them here.

**Comment by godbyk** [comment]
A support ticket is a better channel for issues unrelated to the SteamVR-OSVR software itself. This issue tracker is only for the latter.

**Comment by Balderick** [comment]
Could a support ticket help in getting the steamvr-osvr team to drop a couple of gears?

**Comment by godbyk** [comment]
From reading what you've been reporting here, I thought you'd already determined that yourself.

I'll be honest, though, this issue report has meandered quite a bit and I've lost track of what the current problem is that you're experiencing.

**Comment by Balderick** [comment]
;-)
I can not prove or confirm the multi GPU configuration, the primary display management, the orientation or broken ir tracking are osvr hardware, osvr software, steam, steamvr or operating system issue/s.

**Comment by godbyk** [comment]
While we generally try to build a new SteamVR-OSVR plugin when we rebuild OSVR-Core specifically to avoid this confusion, it's not actually necessary. The interface to the Core libraries rarely changes and the SteamVR-OSVR driver will work with almost any OSVR server version you can get your hands on. (The latest SteamVR-OSVR plugin does require a fairly modern OSVR-RenderManager, though.)

**Comment by Balderick** [comment]
ahh k thanks for cl;arifying that.

I use runtime installer which installs render manager components but not the full set of rendermanager tools.    Is it safe to assume the latest versiion of osvr runtime uses the latest version render manager components?

Can you confirm the example game of TF2 on steamvr-osvr readme.md is even relevant now adays?
All i get is a dialogue box saying something like "TF2 is not a vr game.  Trying to run it in vr mode will drastically reduce performance."   that is with and without -vr launch parameter for that game.

**Comment by godbyk** [comment]
> Is it safe to assume the latest version of osvr runtime uses the latest version render manager components?

Yes.

> As an nvidia shield (tablet and adroid tv) device user the android support of hdk has me intrigued 

Android support is still a work-in-progress. You can follow its development by monitoring the [OSVR-Android-Build repository](https://github.com/osvr/OSVR-Android-Build).

> Can you confirm the example game of TF2 on steamvr-osvr readme.md is even relevant nowadays?

I haven't tried it personally, so I can't say. You could ask on Reddit or in the Gitter chat rooms, though.

> I dare even say the readme.mds for steamosvr-osvr drivers and render manager are useless. They seem to convey mainly outdated and irrelevant information.

That could very well be. I do intend to update the SteamVR-OSVR README for the next release.

If you've found specific problems in either of the READMEs, please file individual issue reports citing the erroneous or outdated information and we'll get it updated. Or even better, you can [edit the file directly](https://github.com/OSVR/SteamVR-OSVR/edit/master/README.md) and submit the changes for review.

> How do i delete the saved beacon configs?

The filename is set in the OSVR server config file (under the `calibrationFile` key). By default it's a file named `videotrackerCombinedCalibrationFile13.json` or similar based on the HDK version. For more information on the calibration utility, you'll need to either contact OSVR support or read through the relevant code and docs in the OSVR-Core repository. I don't know much about it myself.

> [...] the green light on camera has never litten up when osvr server is started

I believe the camera light turns green once the video-based tracker plugin is loaded.  Are you seeing any error messages when you start up the OSVR server? Does it say anything about the plugin?

> Why doesn't uninstalling and reinstalling osvr runtime reset video tracking calibration saved configs?

I don't know the answer to that either. I'd suggest contacting support@osvr.org and they can direct your request to someone who knows more about it than I do.

**Comment by Balderick** [comment]
Thanks for the feedback @godbyk it is appreciated.    I really appreciate developers simply do not have the time for hand holding every osvr user step by step with an issue.  I am sure the readme.md's would benefit greatly of being updated.   I should have more free time once i get osvr hdk working so will try to help in contributing to that or at least making suggestions as issues!.

I did make a post listing what i can see when launching osvr server; somewhere here on github recently, and will try to find it ....

I'll add the specific tracking issue in my already opened osvr.org support ticket.

**Comment by Balderick** [comment]
This is what i can see when launching osvr server https://github.com/OSVR/OSVR-Docs/issues/63

**Comment by Balderick** [comment]
Regardless of what config i choose using osvr Config osvr server starts with same .json giving exact same read out in terminal as listed in osvr/osvr-docs issue #63 
Regardless of what config i choose use to assign from command prompt osvr server starts with same .json giving exact same read out in terminal as listed in osvr/osvr-docs issue #63
Regardless of what sample.json or any other .json is saved as "osvr_server_config.json" in main dir  osvr server starts with same .json giving exact same read out in terminal as listed in osvr/osvr-docs issue #63

Uninstalling and reinstalling different runtimes, deleting the main osvr dir in prgram files folder, running registry cleaner are some of the things i have been trying but the usb issues (audio device) seem to be causing other usb device errors.   A ms usb tranciever wireless kb + m has needed to be reinstalled sevral times now too.
No usb 2.0 ports are available on this machine.  All the onboard usb ports and front panel usb ports are all 3.0 or higher.   Checking the mobo users manual it highlights that any usb 2.0 (legacy) device that does not work as expected on a usb 3.0 port needs a firmware update and is a well documented issue with intel z170 chipset.

**Comment by godbyk** [comment]
There are too many issues filed that I'm losing track of what the current problem is.

If you're having a problem starting the OSVR server, I'd suggest filing an issue in the [OSVR-Core repository](https://github.com/OSVR/OSVR-Core/issues/new) and closing the other issues. Be sure to include the relevant information (system configuration, HDK version, configuration files used, server output, etc.) in the new issue. You can reference these other issue reports you've created, but please provide all the useful information in the new issue so that those of us trying to provide support don't have to jump around and read long threads of (possibly now-irrelevant) information to try to determine what the problem is.

**Comment by Balderick** [comment]
https://github.com/OSVR/SteamVR-OSVR/issues/71

**Comment by Balderick** [comment]
Are the known usb device firmware issues being hugely underestimated?

**Comment by Balderick** [comment]
one of the links i shared was not going to where it was intended to issue #63 should have been https://github.com/OSVR/OSVR-Docs/issues/63

**Comment by godbyk** [comment]
The output in https://github.com/OSVR/OSVR-Docs/issues/63 says, in part:

```
[OSVR Server] Using default config file - pass a filename on the command line to use a different one.
[OSVR Server] Using config file 'osvr_server_config.json'
```

This means that `osvr_server` isn't recognizing that you're passing a different configuration file to it and it's using the default as per usual.

How are you running `osvr_server`?  If you're running it from the command line, you'd type:
```sh
> osvr_server.exe path/to/my/osvr_server_config.json
```

If you try this, what does the server output say (in its entirety)?

**Comment by Balderick** [comment]
how many times do i need to tell you.?
I have tried command line using all available .jsons, manualy saving sample configs as osvr_server_config.json etc etc with three osvr runtime builds two steamvr-osvr driver versions, four different gpu drivers etc etc

When i said all available jsons i meant from all command line options to launch osvr server, all sample and other configs in osvr config, all the sample and other .jsonss in osvr main dir *all give the same server output regardless of how it is launched or which .json is used*

**Comment by godbyk** [comment]
I understand you're frustrated. But I can troubleshoot this without your help. I need to know exactly what you're typing to start the server and exactly what the server it outputting.

I've looked at [the code](https://github.com/OSVR/OSVR-Core/blob/master/apps/osvr_server.cpp#L52) and the only time the server will print `Using default config file` is when `osvr_server.exe` is run without passing in a configuration file on the command line. So I'm trying to reconcile what the code is doing with what you're seeing.

Can you run `osvr_server.exe path/to/osvr_server_config.json` from a Command Prompt and paste everything from the Command Prompt window here? That will show me what directory you're in, what you typed to run the server, and what the server is printing. (When you paste it here, put three backticks on a line by itself before and after the stuff you're pasting [see below]. That'll format it nicely.)

    ```
    paste contents of Command Prompt window
    ```

Thanks!

**Comment by Balderick** [comment]
```
C:\Program Files\OSVR\Runtime\bin>osvr_server.exe sample-configs/osvr_server_config.HDK13ExtendedPortrait.sample
[OSVR Server] Using config file 'sample-configs/osvr_server_config.HDK13ExtendedPortrait.sample'
[OSVR Server]
Could not open config file!
[OSVR Server] Searched in the current directory; file may be misspelled, missing, or in a different directory.

C:\Program Files\OSVR\Runtime\bin>osvr_server.exe sample-configs/osvr_server_config.HDK13ExtendedPortrait.sample.json
[OSVR Server] Using config file 'sample-configs/osvr_server_config.HDK13ExtendedPortrait.sample.json'
[OSVR Server] Constructing server as configured...
[OSVR Server] Loading auto-loadable plugins...
[OSVR Server] Loading plugins...
[OSVR Server]
[OSVR Server] Instantiating configured drivers...
[OSVR] Added device: org_osvr_filter_videoimufusion/HeadFusion
[OSVR] Client context initialized for org.osvr.analysisplugin
[OSVR] Interface initialized for /com_osvr_Multiserver/OSVRHackerDevKitPrediction0/semantic/hmd
[OSVR] Interface initialized for /com_osvr_VideoBasedHMDTracker/TrackedCamera0_0/semantic/hmd/front
[OSVR Server] Successes:
[OSVR Server]  - com_osvr_VideoBasedHMDTracker/VideoBasedHMDTracker
[OSVR Server]  - org_osvr_filter_videoimufusion/VideoIMUFusion
[OSVR Server]
[OSVR Server]
[OSVR Server] Aliases found and parsed from config file.
[OSVR Server] Display descriptor found and parsed from config file.
[OSVR Server] RenderManager config found and parsed from the config file.
[OSVR Server] Triggering automatic hardware detection...
[OSVR Server] Registering shutdown handler...
[OSVR Server] Starting server mainloop: OSVR Server is ready to go!
[OSVR] Performing hardware auto-detection.
[OSVR] Added device: com_osvr_Multiserver/OSVRHackerDevKit0
[OSVR] Added device: com_osvr_Multiserver/OSVRHackerDevKitPrediction0

```

**Comment by godbyk** [comment]
Okay, now we're starting to get somewhere. This is different output than you had posted previous. Notice the line that says:
```text
[OSVR Server] Using config file 'sample-configs/osvr_server_config.HDK13ExtendedPortrait.sample.json'
```

I don't see any errors there. Does the HDK's IR camera light turn green?  If not, do you have any other webcams or imaging devices plugged in to your computer? If you do, then try unplugging those, leaving just the IR camera connected, and then starting the server again. The IR camera light should turn green then.

**Comment by Balderick** [comment]
There is no green light and osvr ir camera is only camera connected.

**Comment by Balderick** [comment]
```
C:\Program Files\OSVR\Runtime\bin>osvr_server.exe
[OSVR Server] Using default config file - pass a filename on the command line to use a different one.
[OSVR Server] Using config file 'osvr_server_config.json'
[OSVR Server] Constructing server as configured...
[OSVR Server] Loading auto-loadable plugins...
[OSVR Server] Loading plugins...
[OSVR Server]
[OSVR Server] Instantiating configured drivers...
[OSVR] Added device: org_osvr_filter_videoimufusion/HeadFusion
[OSVR] Client context initialized for org.osvr.analysisplugin
[OSVR] Interface initialized for /com_osvr_Multiserver/OSVRHackerDevKitPrediction0/semantic/hmd
[OSVR] Interface initialized for /com_osvr_VideoBasedHMDTracker/TrackedCamera0_0/semantic/hmd/front
[OSVR Server] Successes:
[OSVR Server]  - com_osvr_VideoBasedHMDTracker/VideoBasedHMDTracker
[OSVR Server]  - org_osvr_filter_videoimufusion/VideoIMUFusion
[OSVR Server]
[OSVR Server]
[OSVR Server] Aliases found and parsed from config file.
[OSVR Server] Display descriptor found and parsed from config file.
[OSVR Server] RenderManager config found and parsed from the config file.
[OSVR Server] Triggering automatic hardware detection...
[OSVR Server] Registering shutdown handler...
[OSVR Server] Starting server mainloop: OSVR Server is ready to go!
[OSVR] Performing hardware auto-detection.
[OSVR] Added device: com_osvr_Multiserver/OSVRHackerDevKit0
[OSVR] Added device: com_osvr_Multiserver/OSVRHackerDevKitPrediction0
```

**Comment by Balderick** [comment]
![screenshot 31](https://cloud.githubusercontent.com/assets/6255884/16897180/d7498228-4ba1-11e6-938b-e3cb26cfcc64.png)

**Comment by Balderick** [comment]
Just to try and add some info that might be relevant and expand on the answers to the most recent questions from @godbyk have just noticed there is driver issues related to OSVR High-Speed Infrared Tracking Camera in Device Manager.  

Allthough there are no warning s or indication there are driver issues in devices app, control panel devices and printers or device manager; having two audio belt box devices is a new one to me.
Settings > Devices > Connected Devices sees OSVR HDK hardware as 
<img width="276" alt="dev" src="https://cloud.githubusercontent.com/assets/6255884/17296011/f8a6906c-57f6-11e6-84ef-2f066109dee2.PNG">
with Device Manager seeing things like 
<img width="586" alt="devman" src="https://cloud.githubusercontent.com/assets/6255884/17296032/1330766e-57f7-11e6-933c-e1890c388ece.PNG">
and digging into Image devices properties shows 
<img width="299" alt="devprop" src="https://cloud.githubusercontent.com/assets/6255884/17296077/4441a048-57f7-11e6-934d-763037887dab.PNG">

**Comment by Balderick** [comment]
So after two hours to get osvr windows drivers installed for unknown device after a power cycle has now caused some more unexpected behaviour.

Using the latest 2.7 windows drivers pack without hdk connected installs drivers but in add/remove programs three osvr drivers are being used once the hdk is connected and powered up.
Using the latest 2.7 windows driver pack with hdk connected installs five osvr drivers visible in add/emove programs.
Both methods give unknown device/s (sometimes two unknown devices)

Uninstalling osvr drivers using add/remove programs and then rebooting (with hdk disconnected) and then connecting hdk and let windows installs the drivers it wants gets four devices (expected behaviour) showing in control panel and system settings devices with no problem or unknown devices being shown in device manager.

Windows now sees imaging device as a usb camera and yet osvr ir tracking camera is named in devices.
<img width="572" alt="newdevman" src="https://cloud.githubusercontent.com/assets/6255884/17302671/38fc6a60-5815-11e6-9485-2ff0efe802e9.PNG">
<img width="391" alt="newdev2" src="https://cloud.githubusercontent.com/assets/6255884/17302683/442001fe-5815-11e6-94e1-6683487d8406.PNG">

**Comment by Balderick** [comment]
Still having issues with device drivers but would like to confirm firmware version 1.96 has helped a lot with camera issue/s.

**Comment by Balderick** [comment]
Upgrading firmware to 1.96 from windows 1607 fixed the camera.

Still hitting 108, 400 and other steamvr errors.   Have tried recent builds of osvr runtime and steamvr-osvr drivers and the AIO.   All give similar results.

**Comment by godbyk** [comment]
@Balderick Can you upload a system report from SteamVR using the latest build of SteamVR-OSVR?

When I've seen error 108 (HMD not found), it's been because the SteamVR-OSVR driver failed to connect to the OSVR server or if there's been some hardware or driver issue.

Usually error 400 (failed to start compositor) is just a result of error 108.

**Comment by Balderick** [comment]
Depends what caused the 108 ...

**Comment by Balderick** [comment]
Will do

**Comment by godbyk** [comment]
The extended-mode and direct-mode configuration files are only used by RenderManager. Since SteamVR uses its own direct-mode code, it doesn't make use of those files. (The SteamVR-OSVR driver does still make use of those files in part when operating in direct mode, though, since we can't autodetect all the parameters we need in direct mode yet.)

The `steamvr.txt` log looks like it found the HDK in extended mode okay but then had some difficulty when trying to switch to direct mode. (At least that's my guess.)

The `steamvr2.txt` log looks like it's in direct mode and that you ran into the bug I just (hopefully) fixed. (Notice the crazy rotation, position, and vsync values in the vrserver.txt file? Those should be fixed in the latest build.)

Let's look at the vrserver.txt log from the latest build and see where that gets us. It seems like it's pretty close to working.

**Comment by Balderick** [comment]
System report from latest build runtime and steamvr osvr drivers, enabling direct mode and then launching server with default .json - 
[steamvr3.txt](https://github.com/OSVR/SteamVR-OSVR/files/428583/steamvr3.txt)

**Comment by Balderick** [comment]
Just an observation but must say the 1.96 firmware and 1607 build windows seems to have made great advances regarding all the usb device issues i was having.

**Comment by godbyk** [comment]
Yes, @rpavlik has been making great strides in the firmware updates!

I just pushed yet another build (255) that should fix a couple more bugs when the HDK is in direct mode. Can you test that build for me? Before you do, delete the `vr*.txt` log files so we can start fresh. (Easier for me to see which logs are new that way.)

**Comment by Balderick** [comment]
Made sure steamvr.vrsettings and system reports are deleted.    

System report from latest build runtime and 255 build steamvr osvr drivers, enabling direct mode and then launching server with default .json -
[vrserver.txt](https://github.com/OSVR/SteamVR-OSVR/files/428624/vrserver.txt)
vr monitor showed in top left corner of main display with the 108 error. No compositor in hmd.

**Comment by godbyk** [comment]
This log says, `osvr: OSVRTrackedDevice::Activate(): Display startup timed out!`  Check to make sure you have the OSVR server running before you start SteamVR. (If it's been running for a bit, you might shut it down and restart it.)

**Comment by Balderick** [comment]
[vrserver.txt](https://github.com/OSVR/SteamVR-OSVR/files/428640/vrserver.txt)

**Comment by godbyk** [comment]
Hmm.. we're still getting that message.

* When the OSVR server running, also run `osvr_print_tree` and paste the results here.
* Does the OSVR Tracker Viewer show the axes moving and rotating when you move and rotate the HDK?
* Do you have any firewalls or antivirus running that might block communications with the server?
* Do any 'native' (non-SteamVR) OSVR applications work?

**Comment by Balderick** [comment]
1. Using osvr central to print tree another command prompt window shows but disappears before i can read or copy text.  

tbc

**Comment by godbyk** [comment]
The firmware upgrade only impacts the HDK. It's not a camera firmware upgrade.

Try running the `osvr_print_tree` program from a command prompt:

1. Hold down the **Windows** key and press **R**. Type `cmd` and press **Enter**.
2. At the command prompt, change to the location where you installed OSVR. For example: `cd "C:\Program Files\OSVR\Runtime\bin"` or `cd "C:\Program Files\OSVR\SDK\bin"`. Don't forget the quotation marks.
3. Type `osvr_print_tree.exe` and press **Enter**.
4. If you can see output from the program, you can save it to a file: `osvr_print_tree.exe > osvr_print_tree.log`. Then you can upload the `osvr_print_tree.log` file.

**Comment by Balderick** [comment]
[osvr_print_tree.txt](https://github.com/OSVR/SteamVR-OSVR/files/428648/osvr_print_tree.txt)

**Comment by Balderick** [comment]
Here is another system report after unblocking the downloaded zip.   Still have hard to decribe stuff goiing on but basically vr monitor gui
<img width="175" alt="vrmonpng" src="https://cloud.githubusercontent.com/assets/6255884/17835157/f8c58800-6755-11e6-92aa-d378ab2c84a9.PNG">
 is fubar.  
[vrserver.txt](https://github.com/OSVR/SteamVR-OSVR/files/428652/vrserver.txt)

I'll be back

**Comment by godbyk** [comment]
Oh, weird! I've never seen the SteamVR monitor do that. You might right-click on SteamVR in the library and run an integrity check. Or just uninstall/delete it entirely and reinstall a fresh copy. 

Also, let's delete the entries from `steamvr.vrsettings` that have to do with direct mode. Let's see if we can get it running in extended mode first.

**Comment by Balderick** [comment]
Found windows was using work group so set up home network.    After installing the latest runtime i tried tracker viewer and video calibration utility.

Tracker viewer does not detect rotational or positional tracking.   Camera does not get switched on.  Command prompt output gives `C:\Program Files\OSVR\Runtime\bin>OSVRTrackerView.exe
Aug 24 19:49:16.656 notice [OSVR]: Log file created in C:\Users\Tron\AppData\Local\OSVR\Logs
Aug 24 19:49:16.656 notice [OSVR]: Log file name starts with "OSVRTrackerView"
Aug 24 19:49:16.672 notice [OSVR]: Logging for C:\Program Files\OSVR\Runtime\bin\OSVRTrackerView.exe
Aug 24 19:49:16.765 info [OSVR: org.osvr.trackerview]: OSVR client context initialized for org.osvr.trackerview


[TrackerViewer] No arguments passed: default is as if you passed the following:
 --pose /me/hands/left --pose /me/hands/right --pose /me/head
You can specify --pose or --orientation then a path, as many times as you want.
Pass the argument --help for more info.

Aug 24 19:49:18.028 info [OSVR]: Z:\j\workspace\OSVR-Core-Build-Win\BIT\64\TRACING\OFF\VS\12\host\windows\src\osvr\Common\ClientInterface.cpp:38: Interface initialized for /me/hands/left
Aug 24 19:49:18.028 info [OSVR]: Z:\j\workspace\OSVR-Core-Build-Win\BIT\64\TRACING\OFF\VS\12\host\windows\src\osvr\Common\ClientInterface.cpp:38: Interface initialized for /me/hands/right
Aug 24 19:49:18.028 info [OSVR]: Z:\j\workspace\OSVR-Core-Build-Win\BIT\64\TRACING\OFF\VS\12\host\windows\src\osvr\Common\ClientInterface.cpp:38: Interface initialized for /me/head
Aug 24 19:49:17.982 notice [OSVR: org.osvr.trackerview]: Could not connect to OSVR server in the timeout period allotted of 200ms
Aug 24 19:49:18.028 info [OSVR: org.osvr.trackerview]: Could not resolve source for /me/hands/left
Aug 24 19:49:18.028 info [OSVR: org.osvr.trackerview]: Could not resolve source for /me/hands/right
Aug 24 19:49:18.028 info [OSVR: org.osvr.trackerview]: Could not resolve source for /me/head
vrpn_Endpoint::send_pending_reports(): No TCP connection
vrpn_Connection_IP::send_pending_reports:  Closing failed endpoint.

C:\Program Files\OSVR\Runtime\bin>`

Video tracking calibration utility does switch on camera and can see the leds on hdk hmd faceplate; again now.  Tracking is very jittery though and then camera/video sync freezes .     On initial locking of leds the numbers for beacons all appear where they are expected to (where the leds are) but then it is like they all get "sucked" into a small area and the camera can never tell them all apart again.
<img width="483" alt="trackutil2" src="https://cloud.githubusercontent.com/assets/6255884/17945556/477fea1a-6a3c-11e6-85af-ddc406cf1473.PNG">

**Comment by godbyk** [comment]
This message from Tracker Viewer:

    Aug 24 19:49:17.982 notice [OSVR: org.osvr.trackerview]: Could not connect to OSVR server in the timeout period allotted of 200ms

sounds like it can't connect to the OSVR server, which is the same sort of error we're seeing in the SteamVR-OSVR log files.

Either the server isn't running (which I don't think is the case, since we've double-checked that in the past), or there's some problem communicating with it. If we can't communicate with the server, it's because either the server isn't listening for incoming communication requests or because there's something preventing those requests from getting to the server (or responses from the server getting to the client).

When you ran Tracker Viewer, did the OSVR server log show a message about a new client being connected? 

While the server is running, please run `netstat -a` at a command prompt and paste the output here. That will show us what services are listening on which ports. We'll see if the OSVR server is in that list.

**Comment by Balderick** [comment]
yes a new client is started if osvr server is running when tracker viewer is run.


`Aug 24 22:12:16.123 notice [OSVR]: Log file created in C:\Users\Tron\AppData\Local\OSVR\Logs
Aug 24 22:12:16.123 notice [OSVR]: Log file name starts with "osvr_server"
Aug 24 22:12:16.123 notice [OSVR]: Logging for C:\Program Files\OSVR\Runtime\bin\osvr_server.exe
Aug 24 22:12:16.159 info [OSVR Server]: Using default config file - pass a filename on the command line to use a different one.
Aug 24 22:12:16.159 info [OSVR Server]: Using config file 'osvr_server_config.json'.
Aug 24 22:12:16.159 info [OSVR Server]: Constructing server as configured...
Aug 24 22:12:16.199 info [OSVR Server]: Loading auto-loadable plugins...
Aug 24 22:12:16.409 info [OSVR Server]: Loading plugins...
Aug 24 22:12:16.409 info [OSVR Server]: Instantiating configured drivers...
Aug 24 22:12:16.409 notice [OSVR Server]: Added device: org_osvr_filter_videoimufusion/HeadFusion
Aug 24 22:12:16.487 info [OSVR: org.osvr.analysisplugin]: OSVR client context initialized for org.osvr.analysisplugin
Aug 24 22:12:16.518 info [OSVR]: Z:\j\workspace\OSVR-Core-Build-Win\BIT\64\TRACING\OFF\VS\12\host\windows\src\osvr\Common\ClientInterface.cpp:38: Interface initialized for /com_osvr_Multiserver/OSVRHackerDevKitPrediction0/semantic/hmd
Aug 24 22:12:16.518 info [OSVR]: Z:\j\workspace\OSVR-Core-Build-Win\BIT\64\TRACING\OFF\VS\12\host\windows\src\osvr\Common\ClientInterface.cpp:38: Interface initialized for /com_osvr_VideoBasedHMDTracker/TrackedCamera0_0/semantic/hmd/front
Aug 24 22:12:16.518 info [OSVR Server]: Successes:
Aug 24 22:12:16.518 info [OSVR Server]:  - com_osvr_VideoBasedHMDTracker/VideoBasedHMDTracker
Aug 24 22:12:16.518 info [OSVR Server]:  - org_osvr_filter_videoimufusion/VideoIMUFusion
Aug 24 22:12:16.518 info [OSVR Server]: Aliases found and parsed from config file.
Aug 24 22:12:16.518 info [OSVR Server]: Display descriptor found and parsed from config file.
Aug 24 22:12:16.518 info [OSVR Server]: RenderManager config found and parsed from the config file.
Aug 24 22:12:16.518 info [OSVR Server]: Triggering automatic hardware detection...
Aug 24 22:12:16.518 info [OSVR Server]: Registering shutdown handler...
Aug 24 22:12:16.518 info [OSVR Server]: Starting server mainloop: OSVR Server is ready to go!
Aug 24 22:12:16.518 info [OSVR Server]: Performing hardware auto-detection.
directx_camera_server: Using capture device 'USB Video Device' at path '\\?\usb#vid_0bda&pid_57e8&mi_00#9&314fc184&4&0000#{65e8773d-8f56-11d0-a3b9-00a0c9223196}\global'
Aug 24 22:12:16.659 notice [OSVR Server]: Added device: com_osvr_Multiserver/OSVRHackerDevKit0
Aug 24 22:12:16.659 notice [OSVR Server]: Added device: com_osvr_Multiserver/OSVRHackerDevKitPrediction0
Video-based tracker: Camera turned on!
Opening camera 0
Video-based tracker: NOTE: Beacon calibration filename videotrackerCombinedCalibrationFile13.json was specified, but not found or could not be loaded. This is not an error: This may just mean you have not yet run the optional beacon pre-calibration step.
Aug 24 22:12:17.066 info [OSVR]: Z:\j\workspace\OSVR-Core-Build-Win\BIT\64\TRACING\OFF\VS\12\host\windows\src\osvr\Client\AnalysisClientContext.cpp:76: Got updated path tree, processing
Aug 24 22:12:17.062 notice [OSVR Server]: Added device: com_osvr_VideoBasedHMDTracker/TrackedCamera0_0
vrpn: Connection request received from 127.0.0.1: 127.0.0.1 51847
Aug 24 22:12:18.068 info [OSVR Server]: Sent path tree to clients.
Aug 24 22:12:18.068 info [OSVR]: Z:\j\workspace\OSVR-Core-Build-Win\BIT\64\TRACING\OFF\VS\12\host\windows\src\osvr\Client\TrackerRemoteFactory.cpp:93: Constructed a TrackerHandler for com_osvr_Multiserver/OSVRHackerDevKitPrediction0@localhost:3883 sensor 0
Aug 24 22:12:18.071 info [OSVR Server]: Performing hardware auto-detection.
Aug 24 22:12:18.068 info [OSVR]: Z:\j\workspace\OSVR-Core-Build-Win\BIT\64\TRACING\OFF\VS\12\host\windows\src\osvr\Client\TrackerRemoteFactory.cpp:93: Constructed a TrackerHandler for com_osvr_VideoBasedHMDTracker/TrackedCamera0_0@localhost:3883 sensor 0
Aug 24 22:12:18.068 info [OSVR: org.osvr.analysisplugin]: Successfully produced handler for /com_osvr_Multiserver/OSVRHackerDevKitPrediction0/semantic/hmd
Aug 24 22:12:18.068 info [OSVR: org.osvr.analysisplugin]: Successfully produced handler for /com_osvr_VideoBasedHMDTracker/TrackedCamera0_0/semantic/hmd/front
Aug 24 22:12:18.068 info [OSVR: org.osvr.analysisplugin]: Connected 2 of 2 unconnected paths successfully
Aug 24 22:12:18.142 info [OSVR Server]: Sent path tree to clients.
vrpn: Connection request received from 127.0.0.1: 127.0.0.1 51846
Aug 24 22:12:20.271 info [OSVR Server]: Performing hardware auto-detection.
Aug 24 22:12:20.342 info [OSVR Server]: Sent path tree to clients.
vrpn: Connection request received from 127.0.0.1: 127.0.0.1 51850
Aug 24 22:12:29.098 info [OSVR Server]: Performing hardware auto-detection.
Aug 24 22:12:29.168 info [OSVR Server]: Sent path tree to clients.
vrpn: Connection request received from 127.0.0.1: 127.0.0.1 51852
Video-IMU fusion: Hold still, measuring camera pose.


NOTE: For best results, during tracker/server startup, hold your head/HMD still closer than 0.3 meters from the tracking camera for a few seconds, then rotate slowly in all directions.`

Which gets rotational postitioning working.


And 
[netstat.txt](https://github.com/OSVR/SteamVR-OSVR/files/435705/netstat.txt)

**Comment by godbyk** [comment]
Proto  Local Address          Foreign Address        State
    TCP    127.0.0.1:3883         DESKTOP-JV35IAK:0      LISTENING
    UDP    127.0.0.1:3883         *:*

shows OSVR is listening on the usual ports. 

These messages:
```
Video-based tracker: Camera turned on!
Opening camera 0
Aug 24 22:12:18.068 info [OSVR: org.osvr.analysisplugin]: Successfully produced handler for /com_osvr_Multiserver/OSVRHackerDevKitPrediction0/semantic/hmd
Aug 24 22:12:18.068 info [OSVR: org.osvr.analysisplugin]: Successfully produced handler for /com_osvr_VideoBasedHMDTracker/TrackedCamera0_0/semantic/hmd/front
Aug 24 22:12:18.068 info [OSVR: org.osvr.analysisplugin]: Connected 2 of 2 unconnected paths successfully
Video-IMU fusion: Hold still, measuring camera pose.
NOTE: For best results, during tracker/server startup, hold your head/HMD still closer than 0.3 meters from the tracking camera for a few seconds, then rotate slowly in all directions.
```

make it sound like the camera-based tracker is starting up properly. Does the light on the camera turn green now? If you hold the HDK up to the camera for a minute, it should establish a position and start tracking the position of the HDK using the camera.

**Comment by Balderick** [comment]
The camera light comes on when starting server.  The camera detects hmd but can not establish position.    
    Looks good there.  Hold it there ...

**Comment by godbyk** [comment]
Getting closer then. When it says that, hold it still for a minute. If it doesn't work, move the HMD around slightly to try to get it to lock on better. It can be a bit finicky sometimes.

It should establish the position and tell you what it is once it's locked on.

**Comment by Balderick** [comment]
It did used to work on 1.3 hdk !.    Just seems to be getting intermittent info.   The axis on tracker viewer display moves smooth and as expected considering positional tracking is not working.
```
Aug 24 23:15:31.057 info [OSVR Server]: Sent path tree to clients.
vrpn: Connection request received from 127.0.0.1: 127.0.0.1 52166
Video-IMU fusion: Hold still, measuring camera pose...


NOTE: For best results, during tracker/server startup, hold your head/HMD still closer than 0.3 meters from the tracking camera for a few seconds, then rotate slowly in all directions.



That distance looks good, hold it right there.
```

**Comment by godbyk** [comment]
Do the axes in Tracker Viewer change position or do they stay at the origin and just rotate?

The server should continue to print some more info after `hold it right there`. It'll show you the (x, y, z) coordinates of the HMD.

**Comment by Balderick** [comment]
It should.   What i see is the hmd spinning on the z axis (green)   with blue and red offset about 45* clockwise to true and the positions are not found even after persevering for five mins.

**Comment by godbyk** [comment]
I haven't tested it yet, but you're welcome to give ti a shot. Can't really hurt anything and it's easy to undo (just set it to `1` instead of `0`).

**Comment by Balderick** [comment]
It works for me!. 
```
Video-IMU fusion: Hold still, measuring camera pose..........


Video-IMU fusion: Camera pose acquired, entering normal run mode!
Camera is located in the room at roughly  0.0469146 -0.0189539   0.421734

```

Positional and rotational tracking working in tracker viewer now.  :-)

Will go back to trying videotarackingcalibration utility and then steamvr

**Comment by godbyk** [comment]
Awesome!

If the tracking it working well enough as is, you don't need to run the calibration utility.  I'd just give SteamVR a shot and see how it fares.

**Comment by Balderick** [comment]
Okay, i started osvr server using extended landscape json.  Got green lights for camera and hmd so ran room setup in steamvr. 
[vrserver.txt](https://github.com/OSVR/SteamVR-OSVR/files/435855/vrserver.txt)

I know steamvr needs this to get the info from hdk that it needs.

direct mode testing ...

**Comment by Balderick** [comment]
Have been using this video tutorial for guidance/sanity check.  https://www.youtube.com/watch?v=TCA0CBXmS2Q&feature=youtu.be
Which explains the 308 error being associated with vr server not dieing when quiting steamvr and/or steam.

Tried testing Elite Dangerous but could not find the launch in vr mode option.



My hdk is behaving the best (most predictable) it ever has for sure.  Windows 1607, ver 7 camera firmware, 1.96 oled hdk firmware.  Running latest runtime and steamvr-osvr drivers.

It would appear all the issues have been resolved and/or the recommended workarounds are applying/working.    Happy days.   :-)

**Comment by godbyk** [comment]
Excellent! I'm glad to hear it's (finally) working for you.

**Comment by Balderick** [comment]
Big thank you to you.   

And apologies to you and everybody else.

**Comment by Balderick** [comment]
Been having a ball testing and just noticed theres is 258 steamvr drivers and 1328-264 runtime to check out.   Github seems to be out of sync, somehow.

So have had elite dangerous launch in both direct and extended modes. Still working on getting input for game and trying to use steam controller.      So with 255 steamvr drivers and  1320-263 runtime here is 
[directmode.txt](https://github.com/OSVR/SteamVR-OSVR/files/440653/directmode.txt) and i can see 
<img width="176" alt="directmode" src="https://cloud.githubusercontent.com/assets/6255884/18027611/aa47e762-6c60-11e6-8da2-fdfef0a76842.PNG"> and 
<img width="396" alt="vrerror" src="https://cloud.githubusercontent.com/assets/6255884/18027613/bba333c2-6c60-11e6-8938-a56d116a51f9.PNG">
if i select `YES` steam or steamvr crash and or hmd blackscrteens or display gets switched off.   OSVR Control reports display switched off if left running and connected when running steamvr and osvr server.

If `CANCEL` is selected or if that warning is simply ignored elite dangerous launches as expected.


In extended mode steamvr compositor crashes but game still launches if all steamvr errors are ignored.   
[extndmode.txt](https://github.com/OSVR/SteamVR-OSVR/files/440656/extndmode.txt)
Still get the steamvr disabled warning too.

In general the hdk is a behaving astronomicaly better.   Seems far more solid and predictable.   Tracking poses are being detected when hmd is 1.5 meters away from camera!.    Tracking is spot on and slick with rotational vector enabled for elite dangerous using directmesh json.

Both osvr and steamvr seem to removing there install root directories at uninstall time which is a great timesaver.

Steamvr power saving for hmd is working as well as tutorial and room setup and other steamvr stuff but still not got dashboard working yet.

With each steamvr driver and runtime update my hdk is getting slicker and slicker.   Awesome stuff guys.  Thank you.

That vr server process issue is going to get reported to steamvr devs today.

**Comment by Balderick** [comment]
Just a quick update to confirm what has made my hdk into a vr content munching monster.  https://www.reddit.com/r/OSVR/comments/4zrha0/osvr_screen_black/
